src/ucp/tag/tag_match.inl

```diff
@@ -240,25 +239,26 @@ ucp_request_recv_offload_data(ucp_request_t *req, const void *data,
                                       priv->ssend_ack.sender_tag, recv_flags);
     }
 
-    if (ucs_unlikely(req->status != UCS_OK)) {
-       return req->status;
-    }
-
     /* There is no correct offset in middle headers with tag offload flow.
      * All fragments are always in order - can calculate offset by
      * subtraction of already received data.
      * NOTE: total length of unexpected eager offload message is not known
      * until last fragment arrives, so it is initialized to SIZE_MAX. */
     offset = SIZE_MAX - req->recv.tag.remaining;
 
+    if (ucs_unlikely(req->status != UCS_OK)) {
+        goto out;
+    }
```

Comment:
why need to fix tag offload?

Reply:
because of refactor if `ucp_request_process_recv_data`, which is not bound to any specific API now